src/ops/cli/terraform.py

```diff
@@ -130,6 +132,14 @@ def __init__(self, root_dir, cluster_config_path, cluster_config, inventory_gene
                  execute):
         self.cluster_config_path = cluster_config_path
         self.cluster_config = cluster_config
+        # Check if the cluster_config has a strict requirement of OPS version
+        # But only if 'ops_min_version' is specified. Not all clusters configs enforce this
+        if "ops_min_version" in self.cluster_config.conf["terraform"]:
```

Comment:
1. To keep things more easy to read, what do you think about extracting this code block in a dedicated method `def check_ops_version()` or something similar.
2. It's not really ideal to have this checks in the class constructor (`__init__`). What do you think about having this check in the `run` method?
3.  It's possible that for newer integrations to not have the `terraform` section at all, in which this check would fail because `self.cluster_config.conf["terraform"]` would be None.
What do you think about also adding a check such as `'terraform' in self.cluster_config.conf`

```
if "ops_min_version" in self.cluster_config.conf["terraform"]:

```

For the hierarchical

Reply:
Thanks for the feedback ! Will implement these